bundles/net.tourbook/src/net/tourbook/ui/views/rawData/DialogDeleteTourValues.java

```diff
@@ -0,0 +1,742 @@
+/*******************************************************************************
+ * Copyright (C) 2021 Frédéric Bard
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110, USA
+ *******************************************************************************/
+package net.tourbook.ui.views.rawData;
+
+import de.byteholder.geoclipse.map.UI;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import net.tourbook.Messages;
+import net.tourbook.application.TourbookPlugin;
+import net.tourbook.common.util.ITourViewer3;
+import net.tourbook.common.util.Util;
+import net.tourbook.data.TourData;
+import net.tourbook.database.IComputeTourValues;
+import net.tourbook.database.TourDatabase;
+import net.tourbook.importdata.RawDataManager;
+import net.tourbook.importdata.RawDataManager.TourValueType;
+import net.tourbook.tour.TourEventId;
+import net.tourbook.tour.TourLogManager;
+import net.tourbook.tour.TourManager;
+
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.IDialogSettings;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.jface.dialogs.TitleAreaDialog;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.BusyIndicator;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.DateTime;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Shell;
+
+public class DialogDeleteTourValues extends TitleAreaDialog {
```

Comment:
Deleting all tour markers could be improved by skipping all tours which do not have a marker

![mt delete all tour markers](https://user-images.githubusercontent.com/1283445/115114239-75797a80-9f8e-11eb-8759-f8768bde195d.png)

Reply:
> Deleting all tour markers could be improved by skipping all tours which do not have a marker
> 

I wouldn't want that the user thinks that the tour was "forgotten" so maybe it's better to display everything even if there is no tour marker

Reply:
> 
> 
> Some words are missing, e.g. `...All tours in the...`
> 
> ![mt on all](https://user-images.githubusercontent.com/1283445/115112157-7c9b8b00-9f84-11eb-9a35-a74a0a8e6239.png)

Fixed
![image](https://user-images.githubusercontent.com/25235665/115317379-a5648180-a138-11eb-9b34-2d67ce061fc6.png)

Reply:
>     * The `Deselect All` button should be placed at the right border, like in the Re-import dialog
> 
>     * The value types are sorted alphabetically (with the english language) in the Re-import dialog
> 
> 

Fixed
![image](https://user-images.githubusercontent.com/25235665/115329245-45c5a080-a14f-11eb-9982-303e2c0972ea.png)

Reply:
> 
> 
> > Deleting all tour markers could be improved by skipping all tours which do not have a marker
> 
> I wouldn't want that the user thinks that the tour was "forgotten" so maybe it's better to display everything even if there is no tour marker

Because this update was so slow for me, therefore I suggested to improve it. When you want to keep it as it is, then the performance could still be dramatically be improved by concurrent updates. For me, the db 42->43 was so slow, so I implemented it concurrently which has now a good performance

https://github.com/wolfgang-ch/mytourbook/blob/df98402e4bb417c74301276a260e92439911ea94/bundles/net.tourbook/src/net/tourbook/database/TourDatabase.java#L8003-L8149

Reply:
Ok, I will think about it for a next PR as I already have a lot of changes to go into main.
Also, it is new to me so I will need time to learn concurrency code.

Reply:
I've never run EJB/SQL concurrent save actions before this db update, but it seems to work. Without concurrency, some parts in MT would be very slow, e.g. tour geo search

Reply:
I added it in the readme file. I will do in a next PR. But first, I will educate myself on the `ThreadPoolExecutor` (new to me)

Reply:
When I implemented the Upload to Strava and Suunto, I used [async HTTP requests](https://github.com/wolfgang-ch/mytourbook/blob/8d5d459e2526b72581c76625150448472e0ea53e/bundles/net.tourbook.cloud/src/net/tourbook/cloud/strava/StravaUploader.java#L340) because it made it faster. It was not too complicated

So far the threadpoolExecutor doesn't work because the tour reimport is complex and when saving the modified tour, it creates a lock issue.
![image](https://user-images.githubusercontent.com/25235665/116758285-f55f0600-a9cc-11eb-9598-f88ec78be439.png)

Reply:
It looks like Lucene cannot be run concurrently, you can try to create a lock for `FTSearchManager.updateIndex` that only one thread can run it at once

Reply:
That's a possible solution. That means I have to see each device importer to "lock" any call in each device importer when the database is access/modified. Not an easy task

Reply:
When only the fulltext indexer cannot be run concurrently, then this task may be run async and also not for every tour. I already did some UI update tasks async and to prevent that it runs too often, only the newest requests are executed and older are discarded. 
Search for `__runnableCounter` or `runnableCounter` in the code. I think this could be done similar for the fulltext indexer but not in the display thread, a solution could be a `Executors.newSingleThreadExecutor` which is already used in MT

Reply:
Now that I have implemented a lock, the new roadblock I am encountering is that if the user is re-importing 10+ tours for which the import file is not found, then 10+ windows are displayed like this and I don't think it's a good user experience.
Maybe it's faster when everything goes well but having to click 10+ times is not faster but slower in the end (this case, the 10+ tours are from the same file and the user should only have to click once as in the current release)

![image](https://user-images.githubusercontent.com/25235665/118050193-5ba13c80-b36e-11eb-9087-47cc203a82cf.png)

Reply:
This is fore sure not acceptable, the concurrent version could be run when

![image](https://user-images.githubusercontent.com/1283445/118082594-bf9f2180-b3bd-11eb-9839-b8915db8b14f.png)


is checked?

Reply:
That's a possibility. If I do that, do you think it would be useful to display somewhere (message dialog warning? tooltip?) if concurrency is used so the user knows about it ?

Reply:
My suggestion: 

When "Skip..." is unchecked, then a text is displayed below the checkbox control to inform the user, that this is slowing down the task

Reply:
I see that you did something similar in tooltips:

![image](https://user-images.githubusercontent.com/25235665/118320469-d2b00f80-b4eb-11eb-84ad-ab82524be9da.png)

Reply:
When there is enough space in the UI, I prefer nowadays to put this info not in a hidden tooltip.